optuna/samplers/_tpe/sampler.py

```diff
@@ -470,20 +470,18 @@ def _gmm_log_pdf(
     def _sample_from_categorical_dist(self, probabilities, size):
         # type: (np.ndarray, Tuple[int]) -> np.ndarray
 
-        if probabilities.size == 1 and isinstance(probabilities[0], np.ndarray):
-            probabilities = probabilities[0]
-        probabilities = np.asarray(probabilities)
```

Comment:
Thanks for the good catch! I agree that this conversion of type is not necessary.